### PR TITLE
Fix printing bugs around the gsym argument to Ga

### PR DESCRIPTION
**Pull request by eric-wieser**
~~We should merge #339 first, putting up now to trial CI.~~

See commit messages for an explanation of this change.

### CODE REVIEW

File: galgebra/ga.py [hunk 1]
```diff
@@ -1870,18 +1870,16 @@ def _build_reciprocal_basis(self, gsym):
...
-                    det_str = r'\det\left ( ' + gsym + r'\right ) '
+                if self.coords is None:
+                    gsym_obj = MatrixSymbol(gsym, self.n, self.n)
                 else:
-                    det_str = 'det(' + gsym + ')'
+                    gsym_obj = MatrixFunction(gsym, self.n, self.n)(*self.coords)
                 # Define square of pseudo-scalar in terms of metric tensor
                 # determinant
+                det_g = Determinant(gsym_obj)
+                det_g._assumptions['commutative'] = True  # sympy/sympy#19308
```
Comment (eric-wieser) [inline galgebra/ga.py:1880]: This turns out not to be safe :(

Comment (utensil) [reply galgebra/ga.py:1880]: Why did you need to change this in the first place?

File: galgebra/printer.py [hunk 2]
```diff
@@ -822,6 +819,10 @@ def _print_MatrixBase(self, expr):
 
         return out_str
 
+    def _print_Determinant(self, expr):
+        # sympy `uses |X|` by default, we want `det (X)`
+        return r"\det\left ( {}\right )".format(self._print(expr.args[0]))
```
Comment (utensil) [inline galgebra/printer.py:824]: These two styles are just two styles, does SymPy has an options for it？

Comment (eric-wieser) [reply galgebra/printer.py:824]: No, it does not. We could make a patch upstream to support an option, but for now it's easiest just to add it here.

File: galgebra/ga.py (hunk 1 above)
Comment (eric-wieser) [reply galgebra/ga.py:1880]: Because galgebra assumes any non-commutative term is a basis blade.

Comment (utensil) [reply galgebra/ga.py:1880]: I see

Comment (eric-wieser) [reply galgebra/ga.py]: On second thoughts, we should go ahead with making this change anyway, but in a way that doesn't interfere with non-galgebra determinants.

I've updated this PR to do that, and filed an upstream PR at https://github.com/sympy/sympy/pull/19354 to hopefully eliminate the need for the workaround eventually.

**Pull request merged**